modules/rubiconBidAdapter.js

```diff
@@ -691,6 +843,50 @@ export function determineRubiconVideoSizeId(bid) {
   return utils.deepAccess(bid, `mediaTypes.${VIDEO}.context`) === 'outstream' ? 203 : 201;
 }
 
+export function getPriceGranularity(config) {
+  const granularityMappings = {
+    low: [{max: 5.00, increment: 0.50}],
+    medium: [{max: 20.00, increment: 0.10}],
+    high: [{max: 20.00, increment: 0.01}],
+    auto: [
+      {max: 5.00, increment: 0.05},
+      {min: 5.00, max: 10.00, increment: 0.10},
+      {min: 10.00, max: 20.00, increment: 0.50}
+    ],
+    dense: [
+      {max: 3.00, increment: 0.01},
+      {min: 3.00, max: 8.00, increment: 0.05},
+      {min: 8.00, max: 20.00, increment: 0.50}
+    ]
+  }
+  if (config.getConfig('priceGranularity') === 'custom') {
+    return {ranges: config.getConfig('customPriceGranularity').buckets}
+  } else {
+    return {ranges: granularityMappings[config.getConfig('priceGranularity')]}
+  }
+}
+
+// Function to validate the required video params
+export function hasValidVideoParams(bid) {
+  let isValid = true;
+  // required params and their associated object type
+  var requiredParams = {
+    mimes: '[object Array]',
+    protocols: '[object Array]',
+    maxduration: '[object Number]',
+    linearity: '[object Number]',
+    api: '[object Array]'
+  }
+  // loop through each param and verify it has the correct
+  Object.keys(requiredParams).forEach(function(param) {
+    if (Object.prototype.toString.call(utils.deepAccess(bid, 'mediaTypes.video.' + param)) !== requiredParams[param]) {
+      isValid = false;
+      utils.logError('Rubicon Bid Adapter: mediaTypes.video.' + param + ' is required and must be of type:' + requiredParams[param].substring(requiredParams[param].indexOf(' '), requiredParams[param].indexOf(']')));
+    }
+  })
+  return isValid;
+}
```

Comment:
@harpere @idettman 

This is the main thing I added `hasValidVideoParams`

I want to log each param that is missing, but if they are all missing it will end up like so:

![image](https://user-images.githubusercontent.com/13972794/53774378-0ff7cc00-3ea3-11e9-9859-65aa7abd559a.png)


Should I change this?

Make it one log for the whole function?

Or just change it to a warning?

Reply:
Decided to keep this and just make the function more stable as per Eric's request.